core/math/a_star_grid_2d.cpp

```diff
@@ -441,6 +450,11 @@ bool AStarGrid2D::_solve(Point *p_begin_point, Point *p_end_point) {
 			break;
 		}
 
+		// If we have a limit on traversals, increment and stop once we reach the threshold.
+		if (max_traversals != -1 && traversal_count++ >= max_traversals) {
```

Comment:
We like to avoid incrementing in a check because it makes it hard to read, I'd suggest breaking this up

Reply:
Sounds good, fixed.